### PR TITLE
Cold Reactions and Applejack

### PR DESCRIPTION
**Pull request by FalseIncarnate**
Adds the code functionality to make a chemical reaction only occur when at or below a specified temperature.
- Chemical reactions have a new var named "max_temp" which defines the highest temperature that the reaction can occur at. Above this number, the reaction will not occur.

Adds new reaction for freezing water into ice
- Must be at or below 270K to turn water to ice

Adds two new reagents/drinks: Applejack and Jack Rose cocktail
- Applejack is obtained by freezing cider at or below 270K
 - Applejack is slightly more potent than cider, due to it's alcohol concentration being higher
- Jack Rose is a new mixed drink that is 4 parts Applejack, 1 part Lemon Juice

Only Ice and Applejack utilize the cold reactions at this time, but the functionality is in place for future reactions.
- This should allow coders more flexibility with defining new reactions that may realistically occur at low temperatures.
- This could also be used with stabilized reactions to require that they are cooled rather than heated.

Renamed the existing "required_temp" variable for reactions to "min_temp" to better communicate it's purpose and differentiate it from the new "max_temp" variable for cold reactions.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Yay! So glad you implemented this. Goonchem has this, and despite the fact we don't have much use for it now, this will be really nice once its merged in.

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -64,6 +65,15 @@ datum
...
 */
 
+		ice
+			name = "Ice"
+			id = "ice"
+			result = "ice"
+			required_reagents = list("water" = 1)
+			result_amount = 1
+			max_temp = 273
+			mix_message = "Ice forms as the water freezes."
```
Comment (Fox-McCloud) [inline code/modules/reagents/Chemistry-Recipes.dm]: recommend removing the mix sound from this; ice doesn't...bubble...when mixed.

**Comment by KasparoVy** [comment]
Would that mean that for chemistry departments (research and medchem) the chem heaters would, instead, be turned into 'chemical temperature management systems' to facilitate both heating and cooling? (if, theoretically, certain viable medicines or research chems could be created with low temperatures in the future)
I say this because in the above case, medchem is already jam packed with stuff and lacks room for the cooling machines-- while research lab is in a similar state, though less severe

**Comment by FalseIncarnate** [comment]
@KasparoVy The chem heater, despite what the name implies, can cool things. We don't understand how it works, as any attempts to gain an answer have been responded to with "Bluespace." But yea, a better name would be nice, though perhaps one that is a bit shorter or some form of acronym since your suggestion was quite a mouthful.

@Fox-McCloud I'd have to double check, but I believe making it not play the bubble sound would require some snowflakey checks. Also, given that people tend to stick things in the heater and then return their focus to the other machines, that sound (while not fitting) is useful as an indication that the reaction DID in fact occur and that you should retrieve your beaker,

**Comment by Fox-McCloud** [comment]
@FalseIncarnate 

There's a mix_sound var =p

**Comment by FalseIncarnate** [comment]
Removed the mix_sound from Ice and Applejack reactions, as they both involve freezing water which is typically quiet.

**Comment by Fox-McCloud** [comment]
hush little ice mix, don't say a word...

**Comment by KasparoVy** [comment]
woo CTMS

**Pull request merged**